### PR TITLE
update mix deps & refactor code

### PR DESCRIPTION
**Pull request by alexandrubagu**
update mix deps to new ones and refactor a bunch of code

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -42,17 +42,17 @@ defmodule Guardian.Mixfile do
   end
 
   defp deps do
-    [{:jose, "~> 1.8"},
-     {:phoenix, "~> 1.2.0", optional: true},
-     {:plug, "~> 1.0"},
+    [{:jose, "~> 1.8.1"},
```
Comment (doomspork) [inline mix.exs]: @alexandrubagu do we need to include the patch version?  `1.8` should cover `1.8.x`

File: lib/guardian/claims.ex [hunk 2]
```diff
@@ -150,3 +150,4 @@ defmodule Guardian.Claims do
   defp assign_exp_from_ttl(the_claims, _), do: the_claims
 
 end
+
```
Comment (doomspork) [inline lib/guardian/claims.ex]: Please remove extra whitespace.

File: lib/guardian/claims.ex [hunk 3]
```diff
@@ -98,7 +98,7 @@ defmodule Guardian.Claims do
   end
 
   @doc false
-  def ttl(%{"iat" => iat_v} = the_claims, requested_ttl) do
+  def ttl(the_claims = %{"iat" => iat_v}, requested_ttl) do
```
Comment (doomspork) [inline lib/guardian/claims.ex]: What this change necessary?

File: lib/guardian.ex [hunk 4]
```diff
@@ -358,11 +360,16 @@ defmodule Guardian do
   defp jose_jwk(nil), do: jose_jwk(config(:secret_key) || false)
 
   defp encode_claims(claims) do
-    {headers, claims} = strip_value(claims, "headers", %{})
-    {secret, claims} = strip_value(claims, "secret")
-    {_, token} = secret
+    #tuple {header, claims}
```
Comment (doomspork) [inline lib/guardian.ex]: What is the reasoning behind this change?  Seems unnecessary.

File: lib/guardian.ex [hunk 5]
```diff
@@ -278,12 +278,14 @@ defmodule Guardian do
     else
       params
     end
-    params = stringify_keys(params)
-    {secret, params} = strip_value(params, "secret")
+    stringify_params = stringify_keys(params)
+    
+    #tuple {secret, params}
+    secret_params = strip_value(stringify_params, "secret")
```
Comment (doomspork) [inline lib/guardian.ex]: Why did you remove the pattern matching?

File: mix.exs (hunk 1 above)
Comment (alexandrubagu) [inline mix.exs]: That's right

File: lib/guardian/claims.ex (hunk 2 above)
Comment (alexandrubagu) [inline lib/guardian/claims.ex]: I'm ok with this

**Comment by alexandrubagu** [comment]
After I update credo, I have run ```mix credo``` this is the output:
Image

File: lib/guardian.ex (hunk 5 above)
Comment (doomspork) [inline lib/guardian.ex]: Let's go back to using pattern matching here, it's not clear to me why it was removed in the first place.  In fact the resulting code is worse without it, we're now reliant on using pipe chains within function calls to get the values we want.

File: lib/guardian.ex [hunk 6]
```diff
@@ -278,12 +278,14 @@ defmodule Guardian do
...
 
     try do
-      with {:ok, claims} <- decode_token(jwt, secret),
-           {:ok, verified_claims} <- verify_claims(claims, params),
+      with {:ok, claims} <- decode_token(jwt, secret_params |> elem(0)),
```
Comment (doomspork) [inline lib/guardian.ex]: We do not want to use pipes inside function calls, this is messy.

**Comment by alexandrubagu** [comment]
I've revert changes and set some credo checking rules to false

**Comment by doomspork** [comment]
Thank you @alexandrubagu!  I'll look over this later this evening and we get it merged if we're all good to go 👍 

@hassox feel free to take a peek at this.

**Comment by snewcomer** [comment]
@doomspork Any updates on this?  What about upgrading to `phoenix 1.3.0-rc.0` as well?

**Comment by doomspork** [comment]
@snewcomer I still need to look through this PR but we are not making changes for 1.3rc, please see the previous threads on the topic: https://github.com/ueberauth/guardian/issues/276

File: mix.exs [hunk 7]
```diff
@@ -43,16 +43,16 @@ defmodule Guardian.Mixfile do
 
   defp deps do
     [{:jose, "~> 1.8"},
-     {:phoenix, "~> 1.2.0", optional: true},
-     {:plug, "~> 1.0"},
+     {:phoenix, "~> 1.2.1", optional: true},
```
Comment (doomspork) [inline mix.exs]: @alexandrubagu could we update this to: `{:phoenix, "~> 1.2", optional: true}`?  I believe it'll address most of the 1.3rc1 concerns at the same time 👍

Comment (alexandrubagu) [inline mix.exs]: Done :+1:

**Comment by doomspork** [comment]
Thank you @alexandrubagu!

**Comment by doomspork** [comment]
@hassox / @scrogson any final comments before I merge?

**Comment by doomspork** [comment]
Thank you again @alexandrubagu and apologizes for the delays.

**Comment by alexandrubagu** [comment]
No problem
